File: src/src_core/fmg/core/img/FlagModel.java
package fmg.core.img;

import java.beans.PropertyChangeListener;

import fmg.common.geom.SizeDouble;
import fmg.common.notyfier.NotifyPropertyChanged;

/** Model of the flag image */
public class FlagModel implements IImageModel {

   private SizeDouble _size;
   protected NotifyPropertyChanged _notifier = new NotifyPropertyChanged(this);

   public FlagModel() {
      _size = new SizeDouble(40, 40);
   }

   /** width and height in pixel */
   @Override
   public SizeDouble getSize() { return _size; }
   public void setSize(double widhtAndHeight) { setSize(new SizeDouble(widhtAndHeight, widhtAndHeight)) ; }
   @Override
   public void setSize(SizeDouble size) {
      _notifier.setProperty(_size, size, PROPERTY_SIZE);
   }

   @Override
   public void close() {
      _notifier.close();
   }

   @Override
   public void addListener(PropertyChangeListener listener) {
      _notifier.addListener(listener);
   }
   @Override
   public void removeListener(PropertyChangeListener listener) {
      _notifier.removeListener(listener);
   }

}

File: src/src_core/fmg/core/img/FlagModel.cs
using System;
using System.ComponentModel;
using fmg.common.geom;
using fmg.common.notyfier;

namespace fmg.core.img {

    /// <summary> Model of the flag image </summary>
    public class FlagModel : IImageModel {

        private SizeDouble _size = new SizeDouble(AnimatedImageModelConst.DefaultImageSize, AnimatedImageModelConst.DefaultImageSize);
        private BoundDouble _padding = new BoundDouble(AnimatedImageModelConst.DefaultPadding);
        public event PropertyChangedEventHandler PropertyChanged;
        protected readonly NotifyPropertyChanged _notifier;

        public FlagModel() {
            _notifier = new NotifyPropertyChanged(this, ev => PropertyChanged?.Invoke(this, ev), false);
        }

        /// <summary> width and height in pixel </summary>
        public SizeDouble Size {
            get { return _size; }
            set {
                this.CheckSize(value);
                SizeDouble old = _size;
                if (_notifier.SetProperty(ref _size, value))
                    Padding = this.RecalcPadding(Padding, _size, old);
            }
        }

        public BoundDouble Padding {
            get => _padding;
            set {
                this.CheckPadding(value);
                _notifier.SetProperty(ref this._padding, value);
            }
        }

        public void Dispose() {
            _notifier.Dispose();
            GC.SuppressFinalize(this);
        }

    }

}
